src/Hl7.Fhir.Support/Utility/ReflectionHelper.cs

```diff
@@ -274,45 +267,21 @@ static bool hasPublicInstanceReadAccessor(PropertyInfo p)
 #endif
         }
 
-        public static bool HasDefaultPublicConstructor(Type t)
-        {
-            if (t == null) throw Error.ArgumentNull("t");
-
-            if (t.GetTypeInfo().IsValueType)
-                return true;
-
-            return (GetDefaultPublicConstructor(t) != null);
-        }
+        public static bool HasDefaultPublicConstructor(Type t) =>
+            t.GetTypeInfo().IsValueType || GetDefaultPublicConstructor(t) != null;
 
         internal static ConstructorInfo GetDefaultPublicConstructor(Type t)
         {
-#if NETSTANDARD1_6
-            return t.GetTypeInfo().DeclaredConstructors.FirstOrDefault(s => s.GetParameters().Length == 0 && s.IsPublic && !s.IsStatic);
-#else
             BindingFlags bindingFlags = BindingFlags.Instance | BindingFlags.Public;
-
-            return t.GetConstructors(bindingFlags).SingleOrDefault(c => !c.GetParameters().Any());
-#endif
+            return t.GetTypeInfo().GetConstructors(bindingFlags).SingleOrDefault(c => !c.GetParameters().Any());
         }
 
-        public static bool IsNullableType(Type type)
-        {
-            if (type == null) throw Error.ArgumentNull("type");
-
-            return (type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>));
-        }
+        public static bool IsNullableType(Type type) => type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>);
 
-        public static Type GetNullableArgument(Type type)
-        {
-            if (type == null) throw Error.ArgumentNull("type");
-
-            if (IsNullableType(type))
-            {
-                return type.GetTypeInfo().GenericTypeArguments[0];
-            }
-            else
-                throw Error.Argument("type", "Type {0} is not a Nullable<T>".FormatWith(type.Name));
-        }
+        public static Type GetNullableArgument(Type type) =>
```

Comment:
While we are at it, maybe document this public method as well?

Reply:
I've documented a few extra, but there are tons in this .cs file and I actually want to get rid of most ;-)